### PR TITLE
Reduce use of pipe in codebase + refactor tests + linting

### PR DESCRIPTION
**Pull request by olivroy**
# Summary

Just some linting around the package

* Refactors some dplyr code into base R (improve performance)
* Linting using lintr::lint()
* regular expressions matching with `fixed = TRUE`
* Less magrittr pipe (performance / readability)

# Checklist

- [x] I understand and agree to the [Code of Conduct](https://www.contributor-covenant.org/version/2/1/code_of_conduct.html).
- [x] I have added [`testthat`](https://github.com/r-lib/testthat) unit tests to [`tests/testthat`](https://github.com/rstudio/gt/tree/master/tests/testthat) for any new functionality.

Note: a `render_unit()` test should go on R CMD check, as I made a crucial mistake, but only test-coverage.yaml (and local testing) saved me

### CODE REVIEW

**Comment by olivroy** [comment]
I have exactly this locally, I targeted especially the utils_xml file!

**Review by olivroy — COMMENTED** [review]
I gotta say I don't really like (and am not super talented) fiddling with this type of code.  (base R data subsetting and string manipulation)

but the performance improvements seem worth it 
``` r
library(magrittr)
bench::mark(
  mtcars[mtcars$vs == 1,],
  mtcars |> dplyr::filter(vs == 1),
  mtcars %>% dplyr::filter(vs == 1)
)
#> # A tibble: 3 × 6
#>   expression                            min  median `itr/sec` mem_alloc `gc/sec`
#>   <bch:expr>                        <bch:t> <bch:t>     <dbl> <bch:byt>    <dbl>
#> 1 mtcars[mtcars$vs == 1, ]             87µs  95.2µs     9687.   15.42KB    15.1 
#> 2 dplyr::filter(mtcars, vs == 1)      806µs 849.6µs     1072.    3.81MB    10.4 
#> 3 mtcars %>% dplyr::filter(vs == 1)   813µs 849.9µs     1076.    7.45KB     8.28
```

<sup>Created on 2024-05-16 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>


For the review, I'd recommend you to take a look at commits individually.

I made mistakes in the commits that resulted in CI failures, so I'd say that these three commits were made in a hurry:
c6fa2a63f60fc8011d219118bca5f3ea08da9332, b153108ac640adfb18f16be4aa2a3c20b0c2b0b4, and 7c92c8df18986eb40f58d50b4842afabb76eac5b, but the rest of the commits should all be good to look at individually!

File: R/info_tables.R [hunk 1]
```diff
@@ -327,43 +327,47 @@ info_currencies <- function(
...
-      dplyr::select(curr_name, dplyr::everything()) %>%
-      dplyr::mutate(value = 49.95) %>%
-      gt()
-
-    for (i in seq(nrow(curr))) {
+    for (i in seq_len(nrow(curr))) {
 
       tab_1 <-
-        tab_1 %>%
         fmt_currency(
```
Comment (olivroy) [inline R/info_tables.R:339]: Not convinced by this fiddling, will it make things harder to expand if needed? I can revert if you want!

**Review by rich-iannone — APPROVED** [review]
Looks great to me, and thank you so much for your hard work on this!

**Comment by rich-iannone** [comment]
I looked through everything in this PR and agree with all of it. Also, didn't know that `select()` and `distinct()` could be replaced by all well-crafted `distinct()`! Will merge once CI fully passes.

**Comment by olivroy** [comment]
Thanks for looking it over!

Yes, but be careful `distinct()` (and arrange) is like `mutate()` (I have been bitten by it before), which means:

```r
# doesn't work
mtcars |>
  distinct(x  = "vs")
# works
mtcars |> 
  select(x = "vs")
mtcars |>
  distinct(x = .data$vs)
mtcars |>
  arrange(.data$vs)
# deprecated
mtcars |>
  select(.data$vs)
# doesn't work
mtcars |>
  distinct(all_of("vs"))
# solution (with across's little friend)
mtcars |>
  distinct(pick(vs)) # works quoted too.
mtcars |> 
  distinct(pick(all_of("vs")))
# pick doesn't allow renaming with selection 
# doesn't work
  mtcars |> 
  distinct(pick(all_of(c(x = "vs"))))
```

The more you know!

**Comment by rich-iannone** [comment]
Wow, that’s potentially dangerous! But I trust you fully with any new-school `distinct()` work. :)

**Pull request merged**